### PR TITLE
Use upstream Pod client in plank

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov skuznets@redhat.com

/assign @fejta
/cc @cjwagner
depends on #10985

### CODE REVIEW

**Comment by fejta** [comment]
please pass tests

**Comment by fejta** [comment]
I vote we first review https://github.com/kubernetes/test-infra/pull/10942

**Comment by stevekuznetsov** [comment]
SGTM, review this and I can rebase when that lands

**Comment by fejta** [comment]
ready for rebase!

File: prow/cmd/plank/main.go [hunk 1]
```diff
@@ -115,30 +116,24 @@ func main() {
...
+		// we can simply overwrite that one and be content that our cluster
+		// mapping is correct for the controller
+		if context == defaultContext {
+			context = kube.DefaultClusterAlias
 		}
+		buildClients[context] = client.CoreV1().Pods(configAgent.Config().PodNamespace)
 	}
 
-	c, err := plank.NewController(kubeClient, pkcs, githubClient, nil, configAgent, o.totURL, o.selector, o.skipReport)
+	c, err := plank.NewController(kubeClient, buildClients, githubClient, nil, configAgent, o.totURL, o.selector, o.skipReport)
```
Comment (fejta) [inline prow/cmd/plank/main.go]: This will need to change

File: prow/cmd/plank/main.go [hunk 2]
```diff
@@ -115,30 +116,24 @@ func main() {
 		logrus.WithError(err).Fatal("Error getting GitHub client.")
 	}
 
-	kubeClient, _, _, err := o.kubernetes.Client(configAgent.Config().ProwJobNamespace, o.dryRun)
+	kubeClient, defaultContext, kubernetesClients, err := o.kubernetes.Client(configAgent.Config().ProwJobNamespace, o.dryRun)
```
Comment (fejta) [inline prow/cmd/plank/main.go]: here

File: prow/plank/controller.go [hunk 3]
```diff
@@ -69,12 +66,12 @@ type syncFn func(pj kube.ProwJob, pm map[string]kube.Pod, reports chan<- kube.Pr
...
-	pkcs   map[string]kubeClient
-	ghc    GitHubClient
-	log    *logrus.Entry
-	ca     configAgent
-	totURL string
+	kc           kubeClient
+	buildClients map[string]corev1.PodInterface
+	ghc          GitHubClient
+	log          *logrus.Entry
+	ca           configAgent
```
Comment (fejta) [inline prow/plank/controller.go]: here

File: prow/plank/controller.go [hunk 4]
```diff
@@ -92,24 +89,20 @@ type Controller struct {
 }
 
 // NewController creates a new Controller from the provided clients.
-func NewController(kc *kube.Client, pkcs map[string]*kube.Client, ghc GitHubClient, logger *logrus.Entry, ca *config.Agent, totURL, selector string, skipReport bool) (*Controller, error) {
+func NewController(kc *kube.Client, buildClients map[string]corev1.PodInterface, ghc GitHubClient, logger *logrus.Entry, ca *config.Agent, totURL, selector string, skipReport bool) (*Controller, error) {
```
Comment (fejta) [inline prow/plank/controller.go]: here

File: prow/plank/controller.go [hunk 5]
```diff
@@ -92,24 +89,20 @@ type Controller struct {
...
-		ca:          ca,
-		pendingJobs: make(map[string]int),
-		totURL:      totURL,
-		selector:    selector,
-		skipReport:  skipReport,
+		kc:           kc,
+		buildClients: buildClients,
+		ghc:          ghc,
+		log:          logger,
+		ca:           ca,
```
Comment (fejta) [inline prow/plank/controller.go]: here

**Comment by stevekuznetsov** [comment]
Eek. Bad dev env. Anyway, should be good

File: prow/plank/controller.go [hunk 6]
```diff
@@ -66,12 +63,12 @@ type syncFn func(pj kube.ProwJob, pm map[string]kube.Pod, reports chan<- kube.Pr
...
 // Controller manages ProwJobs.
 type Controller struct {
-	kc     kubeClient
-	pkcs   map[string]kubeClient
-	ghc    GitHubClient
-	log    *logrus.Entry
-	config config.Getter
-	totURL string
+	kc           kubeClient
+	buildClients map[string]corev1.PodInterface
```
[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply prow/plank/controller.go]: Oh you can bet your sweet butt it is. I love me some `kc` `pkcg` `jpa` `lksjdfhsdjhf`

[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply prow/plank/controller.go]: Nah when they get too long we will just use obscure Greek translations

[the remark this answers is not in the document]
Comment (fejta) [reply prow/plank/controller.go]: That has the downnside of remaining comprehensible to the ancient Greek speaker

[the remark this answers is not in the document]
Comment (fejta) [reply prow/plank/controller.go]: We should make sure to abbreviate the ancient Greek too

[the remark this answers is not in the document]
Comment (guineveresaenger) [reply prow/plank/controller.go]: 🤣 
I was gonna sic my dad on y'all

**Pull request merged**